third_party/txt/src/txt/directory_asset_data_provider.cc

```diff
@@ -3,16 +3,44 @@
 // found in the LICENSE file.
 
 #include "txt/directory_asset_data_provider.h"
-#include <dirent.h>
 #include <sstream>
 #include "lib/fxl/logging.h"
 #include "third_party/skia/include/core/SkStream.h"
 #include "third_party/skia/include/core/SkTypeface.h"
+#include "utils/WindowsUtils.h"
+
+#if !defined(_WIN32)
+#include <dirent.h>
+#endif
 
 namespace txt {
 
 DirectoryAssetDataProvider::DirectoryAssetDataProvider(
     const std::string& directory_path) {
+#if defined(_WIN32)
+  std::string path = directory_path + "\\*";
+  WIN32_FIND_DATAA ffd;
+  HANDLE directory = FindFirstFileA(path.c_str(), &ffd);
+  if (directory == INVALID_HANDLE_VALUE) {
+    return;
+  }
+
+  do {
+    if ((ffd.dwFileAttributes & FILE_ATTRIBUTE_ARCHIVE) == 0) {
```

Comment:
what is the purpose of this check?

Reply:
This checks that the WIN32_FIND_DATAA isn't a directory or an archive since we're only interested in files. Actually, this should probably be FILE_ATTRIBUTE_DIRECTORY.

Reply:
Since font files are archivable, we only need to consider files which have this attribute.